bitcoin/src/taproot/mod.rs

```diff
@@ -1375,22 +1369,27 @@ impl std::error::Error for TaprootBuilderError {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         use TaprootBuilderError::*;
 
-        match self {
-            InvalidMerkleTreeDepth(_) | NodeNotInDfsOrder | OverCompleteTree | EmptyTree => None,
+        match *self {
+            InvalidMerkleTreeDepth(ref e) => Some(e),
+            NodeNotInDfsOrder | OverCompleteTree | EmptyTree => None,
         }
     }
 }
 
+impl From<InvalidMerkleTreeDepthError> for TaprootBuilderError {
+    fn from(e: InvalidMerkleTreeDepthError) -> Self { Self::InvalidMerkleTreeDepth(e) }
+}
+
 /// Detailed error type for taproot utilities.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
 pub enum TaprootError {
     /// Proof size must be a multiple of 32.
     InvalidMerkleBranchSize(usize),
     /// Merkle tree depth must not be more than 128.
-    InvalidMerkleTreeDepth(usize),
+    InvalidMerkleTreeDepth(InvalidMerkleTreeDepthError),
     /// The last bit of tapleaf version must be zero.
-    InvalidTaprootLeafVersion(u8),
+    InvalidTaprootLeafVersion(InvalidTaprootLeafVersionError),
     /// Invalid control block size.
     InvalidControlBlockSize(usize),
```

Comment:
Might be nice to newtype the other variants too.

Reply:
Added an additional patch, good reviewing - thanks.